test/unit/test_vis.py

```diff
@@ -0,0 +1,301 @@
+import os
+
+import pytest
+
+from deepreg.vis import main, string_to_list
+
+
+def test_string_to_list():
+    string = "abc, 123, def, 456"
+    got = string_to_list(string)
+    expected = ["abc", "123", "def", "456"]
+    assert got == expected
+
+
+def test_gif_slices():
+    """Covered by test_main"""
+    pass
+
+
+def test_tile_slices():
+    """Covered by test_main"""
+    pass
+
+
+def test_gif_warp():
+    """Covered by test_main"""
+    pass
+
+
+def test_gif_tile_slices():
+    """Covered by test_main"""
+    pass
+
+
+def test_main():
+    img_paths = "./data/test/nifti/unit_test/moving_image.nii.gz"
```

Comment:
actually here we should refactor the test

Reply:
@s-sd would you like to refactor the tests in this file a bit? using parameterize ^^

Reply:
@mathpluscode, I was just trying to use parameterize here but it is proving to be quite difficult becuase every test uses a different combination of arguments in main and the checks are also different (sometimes an exception is checked, sometimes a file output, sometimes multiple file outputs, and sometimes a partial string in a file name). So the number of arugments for main and the arguments themselves are different between tests (for example one argument applicable to mode 1 is not applicable to mode 2 etc) and also the checks that are being performed are different between tests. This is becuase of the different modes of the main funcion. I think it might be easier to leave this without parametrize. What do you think?

Reply:
@s-sd In that case, maybe we can change `test_main` to a class `TestMain` and split each test case into separate funcs.

Parameterizing or not is not that important, but making tests independent is necessary.
(I will underline this in the docs later)

Reply:
@s-sd check #416 ;) plz

Reply:
@mathpluscode, thank you! yeah makes sense. I will refactor the tests to fit the description in Issue #416

Reply:
@mathpluscode, I've refactored the tests